File: dotnet/cs/example/Samples/Wav.cs
/*
 * File: Wav.cs
 * Project: Samples
 * Created Date: 13/09/2023
 * Author: Shun Suzuki
 * -----
 * Last Modified: 14/09/2023
 * Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
 * -----
 * Copyright (c) 2023 Shun Suzuki. All rights reserved.
 * 
 */

using AUTD3Sharp;
using AUTD3Sharp.Gain;
using AUTD3Sharp.Modulation.AudioFile;
using AUTD3Sharp.Utils;

namespace Samples;

internal static class WavTest
{
    public static void Test(Controller autd)
    {
        var config = new Silencer();
        autd.Send(config);

        var m = new Wav("sin150.wav");
        var g = new Focus(autd.Geometry.Center + new Vector3d(0, 0, 150));
        autd.Send((m, g));
    }
}

File: python/example/samples/wav.py
"""
File: wav.py
Project: samples
Created Date: 14/09/2023
Author: Shun Suzuki
-----
Last Modified: 14/09/2023
Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
-----
Copyright (c) 2023 Shun Suzuki. All rights reserved.

"""


from pathlib import Path

import numpy as np

from pyautd3 import Controller, Silencer
from pyautd3.gain import Focus
from pyautd3.modulation.audio_file import Wav


async def wav(autd: Controller) -> None:
    config = Silencer()
    await autd.send_async(config)

    f = Focus(autd.geometry.center + np.array([0.0, 0.0, 150.0]))
    m = Wav(Path(__file__).parent / "sin150.wav")

    await autd.send_async((m, f))
